rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -127,38 +145,99 @@ async def get_node(node_id: str) -> Node | None:
     return Node.from_db(node) if node else None
 
 
+# TODO: Delete this
 async def get_graph_ids() -> list[str]:
     return [
-        graph.id
+        graph.graph_id
         for graph in await AgentGraph.prisma().find_many(where={"is_template": False})
     ]  # type: ignore
 
 
-async def get_template_meta() -> list[GraphMeta]:
-    templates = await AgentGraph.prisma().find_many(
-        where={"is_template": True},
+async def get_graphs_meta(
+    is_template: bool = False, is_active: bool = True, latest_version: bool = True
+) -> list[GraphMeta]:
+    """
+    Retrieves graph metadata based on the provided parameters.
+    Defauly behaviour is to get the latest active graph for each graph_id.
+
+    Args:
+        is_template (bool): Indicates whether the graph is a template.
+        is_active (bool): Indicates whether the graph is active.
+        latest_version (bool): Indicates whether to retrieve only the latest version of each graph.
+
+    Returns:
+        list[GraphMeta]: A list of GraphMeta objects representing the retrieved graph metadata.
+    """
+    where_clause = {"is_template": is_template}
+
+    if not is_template:
+        where_clause["is_active"] = is_active
+
+    where_clause = prisma.types.AgentGraphWhereInput(**where_clause)  # type: ignore
+    order_by = {"version": "desc"}  # type: ignore
+
+    graphs = await AgentGraph.prisma().find_many(
+        where=where_clause,
+        distinct=["graph_id"] if latest_version else None,
+        order=order_by,  # type: ignore
```

Comment:
Without `# type: ignore`:
```suggestion
        order={"version": "desc"},
```